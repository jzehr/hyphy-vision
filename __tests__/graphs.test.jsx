"use strict";

jest.unmock("../src/jsx/components/graphs.jsx");

import React from "react";
import renderer from "react-test-renderer";
import {
  DatamonkeySeries,
  DatamonkeyScatterplot,
  DatamonkeyGraphMenu
} from "../src/jsx/components/graphs.jsx";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { mount } from "enzyme";

var _ = require("underscore");

Enzyme.configure({ adapter: new Adapter() });

describe("DatamonkeySeries", () => {
  var x = _.range(10);
  var y = [_.range(10)];

  const component = renderer.create(
    <DatamonkeySeries
      x={x}
      y={y}
      y_label={"y values"}
      marginLeft={50}
      width={500}
      transitions={true}
      doDots={true}
    />
  );

  it("should create an svg element", () => {
    let chart = component.toJSON();
    expect(chart).toMatchSnapshot();
  });
});

describe("DatamonkeyScatterplot", () => {
  var x = _.range(10);
  var y = [_.range(10)];

  const component = renderer.create(
    <DatamonkeyScatterplot
      x={x}
      y={y}
      y_label={"y values"}
      marginLeft={50}
      height={400}
      width={500}
      transitions={true}
      doDots={true}
    />
  );

  it("should create an svg element", () => {
    let chart = component.toJSON();
    expect(chart).toMatchSnapshot();
  });
});

describe("GraphMenu", () => {
  var x_options = ["Site"];
  var y_options = ["alpha", "beta"];

  const updateAxisSelectionMock = jest.fn();

  const component = mount(
    <DatamonkeyGraphMenu
      x_options={x_options}
      y_options={y_options}
      axisSelectionEvent={updateAxisSelectionMock}
    />
  );

  it("should trigger event when selection changes", () => {
    // Click a button
    component.find("a[data-dimension='beta']").simulate("click", {
      target: { dataset: { dimension: "beta", axis: "yaxis" } }
    });
    expect(updateAxisSelectionMock.mock.calls.length).toBe(1);
  });

  it("Graph menu should handle empty y_options", () => {
    const no_ylabel_component = renderer.create(
      <DatamonkeyGraphMenu
        x_options={x_options}
        y_options={[]}
        axisSelectionEvent={updateAxisSelectionMock}
      />
    );

    let menu = no_ylabel_component.toJSON();
    expect(menu).toMatchSnapshot();
  });
});
