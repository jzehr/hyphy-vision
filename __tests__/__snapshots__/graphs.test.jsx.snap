// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatamonkeySeries should create an svg element 1`] = `
<div>
  <svg
    height={400}
    width={500}
  >
    <g
      transform="translate(50,10)"
    />
    <g
      className="hyphy-axis"
      style={null}
      transform="translate(50,370)"
    />
    <g
      className="hyphy-axis"
      style={null}
      transform="translate(45,10)"
    />
    <text
      text-anchor="middle"
      transform="translate(250,396.6666666666667)"
    />
    <text
      text-anchor="middle"
      transform="translate(12.5,200)rotate(-90)"
    />
  </svg>
</div>
`;
